### PR TITLE
Fallback when relation id is not set

### PR DESCRIPTION
**Pull request by m-kuhn**
When the relation reference widget does not have a relation id set, in QField it will show "invalid relation". On QGIS it will fallback to any relation available for this field.
If no configuration is ever done on the attribute form (and only the relation is added in QGIS project properties) this situation occurs.

Fixes https://github.com/opengisch/QField/issues/4049

### CODE REVIEW

File: src/core/utils/relationutils.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include <qgsrelation.h>
+
+class RelationUtils : public QObject
+{
+    Q_OBJECT
+
+  public:
+    explicit RelationUtils( QObject *parent = nullptr );
+
+    Q_INVOKABLE QgsRelation resolveReferencingRelation( QgsProject *qgisProject, QgsVectorLayer *layer, const QString &fieldName, const QString &relationId );
```
Comment (nirvn) [inline src/core/utils/relationutils.h:30]:
```suggestion
    Q_INVOKABLE QgsRelation resolveReferencingRelation( QgsProject *project, QgsVectorLayer *layer, const QString &fieldName, const QString &relationId );
```

File: src/core/utils/relationutils.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#include <qgsproject.h>
+#include <qgsrelationmanager.h>
+#include <qgsvectorlayer.h>
+
+RelationUtils::RelationUtils( QObject *parent )
+  : QObject( parent )
+{
+}
+
+QgsRelation RelationUtils::resolveReferencingRelation( QgsProject *qgisProject, QgsVectorLayer *layer, const QString &fieldName, const QString &relationId )
```
Comment (nirvn) [inline src/core/utils/relationutils.cpp:28]:
```suggestion
QgsRelation RelationUtils::resolveReferencingRelation( QgsProject *project, QgsVectorLayer *layer, const QString &fieldName, const QString &relationId )
```

**Comment by nirvn** [comment]
@m-kuhn , we'd want this in QField 2.7, I was thinking of pushing a 2.7.2 out today or tomorrow (rotation threshold + move feature(s) fix), this would be a welcome addition.

**Review by nirvn — APPROVED** [review]

File: src/core/utils/relationutils.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#include <qgsvectorlayer.h>
+
+RelationUtils::RelationUtils( QObject *parent )
+  : QObject( parent )
+{
+}
+
+QgsRelation RelationUtils::resolveReferencingRelation( QgsProject *project, QgsVectorLayer *layer, const QString &fieldName, const QString &relationId )
+{
+  QgsRelationManager *relationManager = qgisProject->relationManager();
```
Comment (nirvn) [inline src/core/utils/relationutils.cpp:30]:
```suggestion
  QgsRelationManager *relationManager = project->relationManager();
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 745313615b552155104eae38638b9496e2c168eb: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4058-76e48b7b513ea1f645a7c5e746f428e87a610050-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4058-76e48b7b513ea1f645a7c5e746f428e87a610050-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4058-76e48b7b513ea1f645a7c5e746f428e87a610050-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4058-76e48b7b513ea1f645a7c5e746f428e87a610050-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4058-76e48b7b513ea1f645a7c5e746f428e87a610050-arm64-android.apk)
- [all access arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4058-76e48b7b513ea1f645a7c5e746f428e87a610050-arm-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4058-76e48b7b513ea1f645a7c5e746f428e87a610050-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4058-76e48b7b513ea1f645a7c5e746f428e87a610050-x86-android.apk)
</details>

**Pull request merged**